### PR TITLE
Eventhub review

### PR DESCRIPTION
**Pull request by sbenedicadb**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPCore/Sources/eventhub/EventHub.swift [hunk 1]
```diff
@@ -22,13 +22,13 @@ public typealias EventPreprocessor = (Event) -> Event
 /// Responsible for delivering events to listeners and maintaining registered extension's lifecycle.
 final class EventHub {
     private let LOG_TAG = "EventHub"
-    private let eventHubQueue = DispatchQueue(label: "com.adobe.eventhub.queue")
-    private var registeredExtensions = ThreadSafeDictionary<String, ExtensionContainer>(identifier: "com.adobe.eventhub.registeredExtensions.queue")
-    private let eventNumberMap = ThreadSafeDictionary<UUID, Int>(identifier: "com.adobe.eventhub.eventNumber.queue")
-    private let responseEventListeners = ThreadSafeArray<EventListenerContainer>(identifier: "com.adobe.eventhub.response.queue")
+    private let eventHubQueue = DispatchQueue(label: "com.adobe.eventHub.queue")
```
Comment (sbenedicadb) [inline AEPCore/Sources/eventhub/EventHub.swift:25]: consistency w/ using `camelCase` in our identifier strings and queue labels

File: AEPCore/Sources/eventhub/EventHubPlaceholderExtension.swift [hunk 2]
```diff
@@ -12,11 +12,11 @@
 
 import Foundation
 
-/// A "dummy" extension for `EventHub` so it can share state
+/// An `Extension` for `EventHub`. This serves no purpose other than to allow `EventHub` to share state.
 class EventHubPlaceholderExtension: Extension {
     let name = EventHubConstants.NAME
     let friendlyName = EventHubConstants.FRIENDLY_NAME
-    static let extensionVersion = "0.0.1"
+    static var extensionVersion = EventHubConstants.VERSION_NUMBER
```
Comment (nporter-adbe) [inline AEPCore/Sources/eventhub/EventHubPlaceholderExtension.swift:19]: This can't be a `let`?

File: AEPCore/Sources/eventhub/EventHub.swift [hunk 3]
```diff
@@ -239,12 +244,11 @@ final class EventHub {
             }
         }
 
-        // TODO: Determine which version of Core to use in the top level version field
-        let data: [String: Any] = [EventHubConstants.EventDataKeys.VERSION: ConfigurationConstants.EXTENSION_VERSION,
+        let data: [String: Any] = [EventHubConstants.EventDataKeys.VERSION: EventHubConstants.VERSION_NUMBER,
```
Comment (sbenedicadb) [inline AEPCore/Sources/eventhub/EventHub.swift:247]: updated to version number in eventhub constants instead of configuration constants

**Review by nporter-adbe — APPROVED** [review]

File: AEPCore/Sources/eventhub/SharedState.swift [hunk 4]
```diff
@@ -126,8 +126,8 @@ class SharedState {
         ///   - version: the version of the shared state
         ///   - data: the data for the shared state
         ///   - status: the status of the shared state
-        /// - Returns: A `Node?` with a reference to the previous `Node`
-        func append(version: Int, data: [String: Any]?, status: SharedStateStatus) -> Node? {
+        /// - Returns: A `Node` with a reference to the previous `Node`
+        func append(version: Int, data: [String: Any]?, status: SharedStateStatus) -> Node {
```
Comment (sbenedicadb) [inline AEPCore/Sources/eventhub/SharedState.swift:130]: return value is no longer an optional

**Pull request merged**